creator-node/test/snapbackSM.test.js

```diff
@@ -620,36 +614,253 @@ describe('test SnapbackSM', function () {
-    assert.ok(!snapback.enabledReconfigModesSet.has(RECONFIG_MODES.MULTIPLE_SECONDARIES.key))
-    assert.ok(!snapback.enabledReconfigModesSet.has(RECONFIG_MODES.PRIMARY_AND_OR_SECONDARIES.key))
+    assert.ok(snapback.isReconfigEnabled(RECONFIG_MODES.ONE_SECONDARY.key))
+    assert.ok(!snapback.isReconfigEnabled(RECONFIG_MODES.MULTIPLE_SECONDARIES.key))
+    assert.ok(!snapback.isReconfigEnabled(RECONFIG_MODES.PRIMARY_AND_OR_SECONDARIES.key))
 
     nodeConfig.set('snapbackHighestReconfigMode', RECONFIG_MODES.RECONFIG_DISABLED.key)
     snapback = new SnapbackSM(nodeConfig, getLibsMock())
 
     assert.strictEqual(snapback.highestEnabledReconfigMode, RECONFIG_MODES.RECONFIG_DISABLED.key)
-    assert.ok(snapback.enabledReconfigModesSet.has(RECONFIG_MODES.RECONFIG_DISABLED.key))
-    assert.ok(!snapback.enabledReconfigModesSet.has(RECONFIG_MODES.ONE_SECONDARY.key))
-    assert.ok(!snapback.enabledReconfigModesSet.has(RECONFIG_MODES.MULTIPLE_SECONDARIES.key))
-    assert.ok(!snapback.enabledReconfigModesSet.has(RECONFIG_MODES.PRIMARY_AND_OR_SECONDARIES.key))
+    assert.ok(!snapback.isReconfigEnabled(RECONFIG_MODES.ONE_SECONDARY.key))
+    assert.ok(!snapback.isReconfigEnabled(RECONFIG_MODES.MULTIPLE_SECONDARIES.key))
+    assert.ok(!snapback.isReconfigEnabled(RECONFIG_MODES.PRIMARY_AND_OR_SECONDARIES.key))
+  })
+
+  it('[aggregateReconfigAndPotentialSyncOps] if the self node is the secondary and a primary spId is different from what is on chain, issue reconfig', async function () {
+    const snapback = new SnapbackSM(nodeConfig, getLibsMock())
+
+    // Mock these as very nodes that completed only successful syncs
+    snapback._computeUserSecondarySyncSuccessRates = async () => {
+      return {
+        'http://cnWithSpId2.co': {
+          SuccessRate: 100
+        },
+        'http://cnWithSpId3.co': {
+          SuccessRate: 100
+        }
+      }
+    }
+
+    // Mock that one of the nodes got reregistered from spId 3 to spId 4
+    snapback.peerSetManager.endpointToSPIdMap = {
+      'http://cnReregisteredAsSpId4.co': 4,
+      'http://cnWithSpId2.co': 2,
+      'http://cnWithSpId3.co': 3
+    }
+
+    snapback.endpoint = 'http://cnWithSpId2.co'
+
+    const nodeUsers = [{
+      'user_id': 1,
+      'wallet': '0x00fc5bff87afb1f15a02e82c3f671cf5c9ad9e6d',
+      'primary': 'http://cnReregisteredAsSpId4.co',
+      'secondary1': 'http://cnWithSpId2.co',
+      'secondary2': 'http://cnWithSpId3.co',
+      'primarySpID': 1,
+      'secondary1SpID': 2,
+      'secondary2SpID': 3
+    }]
+
+    const unhealthyPeers = new Set()
+
+    const { requiredUpdateReplicaSetOps, potentialSyncRequests } = await snapback.aggregateReconfigAndPotentialSyncOps(nodeUsers, unhealthyPeers)
+
+    // Make sure that the CN with the different spId gets put into `requiredUpdateReplicaSetOps`
+    assert.strictEqual(requiredUpdateReplicaSetOps[0].unhealthyReplicas[0], 'http://cnReregisteredAsSpId4.co')
```

Comment:
don't need to handle but i think we can be more speicifc here and assert that `unhealthyReplicas === ['http://cnReregisteredAsSpId4.co']` right

Reply:
the problem with that is that we have to check the contents of the arr as your example comparison would fail as the addr of the arr does not match the addr of `unhealthyReplicas`

```
  it('test', () => {
    const vicky = ['vicky']
    assert.strictEqual(vicky, ['vicky'])
  })
```

test result:
```
      AssertionError [ERR_ASSERTION]: Values have same structure but are not reference-equal:

[
  'vicky'
]
```

Reply:
thats just bc === doesn't work on arrays, simplest would be to use lodash isEqual but also whatever, not worth